client/test/debug/FabricJavaDebugConfigurationProvider.test.ts

```diff
@@ -0,0 +1,205 @@
+        let readFileStub: sinon.SinonStub;
+        let readJsonStub: sinon.SinonStub;
+        let registryEntry: FabricGatewayRegistryEntry;
+        let getConnectionStub: sinon.SinonStub;
+        let date: Date;
+        let formattedDate: string;
+        let startDebuggingStub: sinon.SinonStub;
+
+        beforeEach(() => {
+            mySandbox = sinon.createSandbox();
+            clock = sinon.useFakeTimers({ toFake: ['Date'] });
+            date = new Date();
+            formattedDate = dateFormat(date, 'yyyymmddHHMM');
+            fabricDebugConfig = new FabricJavaDebugConfigurationProvider();
+
+            runtimeStub = sinon.createStubInstance(FabricRuntime);
+            runtimeStub.getName.returns('localfabric');
+            runtimeStub.getConnectionProfile.resolves({ peers: [{ name: 'peer1' }] });
+            runtimeStub.getChaincodeAddress.resolves('127.0.0.1:54321');
+            runtimeStub.isRunning.resolves(true);
+            runtimeStub.isDevelopmentMode.returns(true);
+
+            registryEntry = new FabricGatewayRegistryEntry();
+            registryEntry.name = 'local_fabric';
+            registryEntry.connectionProfilePath = 'myPath';
+            registryEntry.managedRuntime = true;
+
+            mySandbox.stub(FabricRuntimeManager.instance(), 'getRuntime').returns(runtimeStub);
+            mySandbox.stub(FabricGatewayRegistry.instance(), 'get').returns(registryEntry);
+
+            workspaceFolder = {
+                name: 'myFolder',
+                uri: vscode.Uri.file('myPath')
+            };
+
+            readJsonStub = mySandbox.stub(fs, 'readJSON');
+            readFileStub = mySandbox.stub(fs, 'readFile').resolves(`{
+                "name": "mySmartContract",
+                "version": "0.0.1"
+            }`);
+
+            debugConfig = {
+                type: 'fabric:java',
+                name: 'Launch Program'
+            };
+
+            debugConfig.request = 'myLaunch';
+            debugConfig.cwd = 'myCwd';
+            debugConfig.args = ['--peer.address', 'localhost:12345'];
+
+            findFilesStub = mySandbox.stub(vscode.workspace, 'findFiles').resolves([]);
+
+            commandStub = mySandbox.stub(vscode.commands, 'executeCommand');
+
+            packageEntry = new PackageRegistryEntry();
+            packageEntry.name = 'banana';
+            packageEntry.version = 'vscode-13232112018';
+            packageEntry.path = path.join('myPath');
+            commandStub.withArgs(ExtensionCommands.PACKAGE_SMART_CONTRACT, sinon.match.any, sinon.match.any).resolves(packageEntry);
+            commandStub.withArgs(ExtensionCommands.INSTALL_SMART_CONTRACT, null, sinon.match.any).resolves({
+                name: 'test-package@0.0.1',
+                path: 'some/path',
+                version: '0.0.1'
+            });
+            commandStub.withArgs('blockchainExplorer.connectEntry', sinon.match.any);
```

Comment:
can you make this ExtensionCommands.CONNECT or whatever it is?

Reply:
I sure can!